tests/e2e/hcp_machine_pool_test.go

```diff
@@ -929,6 +964,34 @@ var _ = Describe("HCP MachinePool", ci.Day2, ci.FeatureMachinepool, func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(MatchRegexp(`KubeletConfig with name 'notexisting'[\n\t\s]*does not exist for cluster`))
 		})
+
+		It("imdsv2 fields - [id:75393]", ci.Medium, func() {
+			By("Try to create a nodepool with invalid imdsv2")
+			mpName := helper.GenerateRandomName("np-75393", 2)
+			validateMPArgAgainstErrorSubstrings(mpName, func(args *exec.MachinePoolArgs) {
+				args.Ec2MetadataHttpTokens = helper.StringPointer("invalid")
+			}, "Expected a valid param. Options are [optional required]. Got invalid.")
+
+			By("Get the deafult nodepool")
+			npList, err := cms.ListNodePools(ci.RHCSConnection, clusterID)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(npList).ToNot(BeEmpty())
+
+			defaultNP := npList[0]
+			Expect(defaultNP.ID()).ToNot(BeNil())
+
+			By("Try to edit ec2_metadata_http_tokens")
+			otherHttpToken := constants.DefaultEc2MetadataHttpTokens
+			if string(defaultNP.AWSNodePool().Ec2MetadataHttpTokens()) == constants.OptionalEc2MetadataHttpTokens {
+				otherHttpToken = constants.RequiredEc2MetadataHttpTokens
+			}
+			errMsg := fmt.Sprintf(
+				`Attribute aws_node_pool.ec2_metadata_http_tokens, cannot be changed from "%s" to "%s"`,
+				string(defaultNP.AWSNodePool().Ec2MetadataHttpTokens()), otherHttpToken)
+			validateMPArgAgainstErrorSubstrings(defaultNP.ID(), func(args *exec.MachinePoolArgs) {
```

Comment:
This will override default nodepool values ... 
Please use the `default-machine-pool` manifests and add a defer to restore values or use another nodepool

Reply:
I'll change it, Thanks